WalletWasabi/Stores/IndexStore.cs

```diff
@@ -398,10 +398,10 @@ private async Task TryCommitToFileAsync(TimeSpan throttle, CancellationToken can
 			}
 		}
 
-		public async Task ForeachFiltersAsync(Func<FilterModel, Task> todo, Height fromHeight)
+		public async Task ForeachFiltersAsync(Func<FilterModel, Task> todo, Height fromHeight, CancellationToken cancel = default)
 		{
-			using (await MatureIndexFileManager.Mutex.LockAsync().ConfigureAwait(false))
-			using (await IndexLock.LockAsync().ConfigureAwait(false))
+			using (await MatureIndexFileManager.Mutex.LockAsync(cancel).ConfigureAwait(false))
```

Comment:
The order of the locks is not consistent and it is one of the code smell that can cause deadlock. IMO it should be `IndexLock`, `MatureIndexFileManager.Mutext` and then `ImmatureIndexFileManager.Mutext`

Reply:
It can be a separate PR. There are other places in the file where the same order was used. So if you want to change this the whole file should be reviewed.